packages/sanity/src/core/theme/_legacy/theme.ts

```diff
@@ -0,0 +1,134 @@
+import {black, blue, gray, green, red, white, yellow} from '@sanity/color'
+import {studioTheme as defaults} from '@sanity/ui'
+import {StudioTheme} from '../types'
+import {buildColor} from './color'
+import {buildFonts} from './fonts'
+import {_isDark} from './helpers'
+import {buildLegacyPalette} from './palette'
+import {buildLegacyTones} from './tones'
+import {LegacyThemeProps} from './types'
+
+/**
+ * Build a Sanity UI theme from legacy CSS properties.
+ *
+ * @example
+ * ```tsx
+ * import {buildLegacyTheme, createConfig} from 'sanity'
+ *
+ * export default createConfig({
+ *   // project configuration ...
+ *
+ *   // Customize theming
+ *   theme: buildLegacyTheme({
+ *     '--black': '#000',
+ *     '--gray': '#777',
+ *     '--focus-color': '#00f',
+ *   })
+ * })
+ * ```
+ *
+ * @public
+ */
+export function buildLegacyTheme(partialLegacyTheme: Partial<LegacyThemeProps>): StudioTheme {
+  const legacyTheme = resolveLegacyTheme(partialLegacyTheme)
+  const THEME_IS_DEFAULT_LEGACY = legacyTheme['--sanity-theme-default']
+  const legacyPalette = buildLegacyPalette(legacyTheme)
+  const legacyTones = buildLegacyTones(legacyPalette)
+  const color = buildColor(legacyPalette, legacyTones)
+  const fonts = buildFonts(legacyTheme)
+
+  return {
+    __dark: _isDark(color.light.default.base.bg, color.light.default.base.fg),
+    __legacy: true,
+    ...defaults,
+    color: THEME_IS_DEFAULT_LEGACY ? defaults.color : color,
+    focusRing: {
+      offset: -1,
+      width: 2,
+    },
+    fonts: THEME_IS_DEFAULT_LEGACY ? defaults.fonts : fonts,
+    media: [
+      parseInt(legacyTheme['--screen-medium-break'], 10) || 512,
+      parseInt(legacyTheme['--screen-default-break'], 10) || 640,
+      parseInt(legacyTheme['--screen-large-break'], 10) || 960,
+      parseInt(legacyTheme['--screen-xlarge-break'], 10) || 1600,
+    ],
+  }
+}
+
+const defaultCustomProperties: LegacyThemeProps = {
+  '--font-family-monospace': defaults.fonts.code.family,
+  '--font-family-base': defaults.fonts.text.family,
+
+  '--black': black.hex,
+  '--white': white.hex,
+
+  // Brand
+  '--brand-primary': blue[500].hex,
+  // '--brand-primary--inverted': white.hex,
+  // '--brand-secondary': red[500].hex,
+  // '--brand-secondary--inverted': white.hex,
+
+  '--component-bg': white.hex,
+  '--component-text-color': black.hex,
+
+  // Gray
+  '--gray': gray[500].hex,
+  '--gray-base': gray[500].hex,
+  // '--gray-darkest': gray[900].hex,
+  // '--gray-darker': gray[800].hex,
+  // '--gray-dark': gray[600].hex,
+  // '--gray': gray[500].hex,
+  // '--gray-light': gray[400].hex,
+  // '--gray-lighter': gray[200].hex,
+  // '--gray-lightest': gray[100].hex,
```

Comment:
What is the deal with these commented variables?

Reply:
They were configured in v2, but by adding TypeScript to this, it seems they're not being used. Will remove commonets.